sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1700,6 +1700,21 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val VIEW_SCHEMA_BINDING = buildConf("spark.sql.defaultViewSchemaBinding")
+    .doc("Control the default behavior of views when the underlying schema changes. " +
+      "Valid values are: " +
+      "'BINDING': Safe casts only. " +
+      "'COMPENSATION': Any supported casts. " +
+      "'TYPE EVOLUTION': Adopt type changes." +
+      "'EVOLUTION': Adopt names, types and arity." +
+      "'UNSUPPORTED': Disable the feature.")
+    .version("4.0.0")
+    .stringConf
+    .transform(_.toUpperCase(Locale.ROOT))
+    .checkValues(Set("BINDING", "COMPENSATION", "TYPE EVOLUTION", "EVOLUTION", "UNSUPPORTED"))
```

Comment:
Let's rename `TYPE EVOLUTION` to `TYPE_EVOLUTION`, so that it is consistent with other values such as `CodegenObjectFactoryMode`

Reply:
If we do this, would we also need to change it in the grammar for CREATE VIEW?